File: legacy-repo/src/main/java/com/clinic/legacy/Patient.java
package com.clinic.legacy;

import java.time.LocalDate;
import java.time.Period;

/**
 * Represents a patient record in the clinic system.
 */
public class Patient {

    private Integer id;
    private String fullName;

    // SEEDED ISSUE (GDPR-032 / HIPAA-164): sensitive personal and health data
    // stored as plain, unencrypted String fields.
    private String socialSecurityNumber;
    private String diagnosis;

    private LocalDate dateOfBirth;

    public Patient(String idStr, String fullName, String socialSecurityNumber,
                   String diagnosis, java.sql.Date dateOfBirth) {
        this.id = Integer.parseInt(idStr);
        this.fullName = fullName;
        this.socialSecurityNumber = socialSecurityNumber;
        this.diagnosis = diagnosis;
        this.dateOfBirth = dateOfBirth.toLocalDate();
    }

    public Integer getId() {
        return id;
    }

    public String getFullName() {
        return fullName;
    }

    public String getSocialSecurityNumber() {
        return socialSecurityNumber;
    }

    public String getDiagnosis() {
        return diagnosis;
    }

    public LocalDate getDateOfBirth() {
        return dateOfBirth;
    }

    /**
     * Returns the patient's age in whole years.
     */
    public int getAgeInYears() {
        return Period.between(dateOfBirth, LocalDate.now()).getYears();
    }

    @Override
    public String toString() {
        // SEEDED ISSUE (GDPR-005): full PHI concatenated into a plain string,
        // encourages data-minimization violations wherever this is logged.
        StringBuilder sb = new StringBuilder();
        sb.append("Patient{id=").append(id)
          .append(", name=").append(fullName)
          .append(", ssn=").append(socialSecurityNumber)
          .append(", diagnosis=").append(diagnosis)
          .append("}");
        return sb.toString();
    }
}

File: legacy-repo-python/clinic/patient.py
"""
Patient domain model for the Clinic Records system.
"""
import datetime


class Patient(object):

    def __init__(self, patient_id, full_name, social_security_number,
                 diagnosis, date_of_birth):
        # patient_id accepted as string and stored as int
        self.id = int(patient_id)
        self.full_name = full_name

        # SEEDED ISSUE (GDPR-032 / HIPAA-164): sensitive personal and health data
        # stored as plain, unencrypted string fields.
        self.social_security_number = social_security_number
        self.diagnosis = diagnosis

        # date_of_birth expected as datetime.date
        self.date_of_birth = date_of_birth

    def get_age_in_years(self):
        """Return the patient's age in whole years.

        SEEDED ISSUE (legacy pattern): uses implicit integer division.
        The correct calculation uses date arithmetic, but this legacy version
        approximates via day-count division — integer division truncates
        unexpectedly in Python 2 (365.25 treated as 365 due to int/int → int).
        """
        today = datetime.date.today()
        days_lived = (today - self.date_of_birth).days
        # LEGACY: implicit integer division — in Python 2, 365 / 365.25 → 0
        # because both operands are treated as int if the literal isn't floated.
        # This produces wrong results for patients under 1 year old and is
        # non-obvious. The correct form is days_lived // 365.25 or date arithmetic.
        age = days_lived / 365.25
        return int(age)

    def __repr__(self):
        # SEEDED ISSUE (GDPR-005): full PHI concatenated into a plain string,
        # encourages data-minimization violations wherever this is logged or printed.
        return (
            "Patient{id=" + str(self.id) +
            ", name=" + self.full_name +
            ", ssn=" + self.social_security_number +
            ", diagnosis=" + self.diagnosis +
            "}"
        )
